onnxruntime/core/mlas/lib/mlasi.h

```diff
@@ -109,7 +109,7 @@ Module Name:
 // Define the maximum number of threads supported by this implementation.
 //
 
-#define MLAS_MAXIMUM_THREAD_COUNT                   16
+#define MLAS_MAXIMUM_THREAD_COUNT                   64
```

Comment:
A couple of things here:

* Should we rename MLAS_MAXIMUM_THREAD_COUNT -> MLAS_MAXIMUM_TASK_COUNT to avoid ambiguity.  [ If so, probably a separate PR just handling the name in the future ]

* Do we have data on the impact on large non-heterogenous systems?  Let's say we have a machine with 32 cores.  Today, the default will be to use 32 threads in the thread pool, but the MLAX_MAXIMUM_THREAD_COUNT=16 will hold back the number of iterations used to just 16.  Ideally, the remaining 16 threads in the pool will remain blocked.  Increasing the maximum will lead us to attempt to parallelize over the full machine, and have all 32 threads active (on sufficiently big tensors).  Not sure whether we expect a benefit there, or if we should set this dynamically based on the CPUID info.